### PR TITLE
Update nix documentation in CONTRIBUTING.adoc and RELEASE.adoc

### PR DESCRIPTION
**Pull request by zeme-wana**
Instructions on how to install & upgrade nix and how to setup the binary caches are now included in the [relevant section](https://github.com/input-output-hk/iogx/blob/main/doc/api.md#flakenix) of the IOGX manual. 
This PR updates the relevant sections of the README/CONTRIBUTING/RELEASE documents.

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -14,87 +14,23 @@ If you _really_ cannot use Nix and still want to contribute to Plutus then xref:
...
 
-This project uses Flakes, so you must have Nix version `>= 2.4` installed.
+The nix code is based on a template from the the 
+[IOGX flake](https://github.com/input-output-hk/iogx). 
 
-Either upgrade or install https://nixos.org/nix/[Nix] following the instructions on https://nixos.org/nix/[its website].
+For instructions on how to install and configure nix (including how to enable 
+access to our binary caches), and how keep the flake inputs up to date, refer to 
+[this section](https://github.com/input-output-hk/iogx/blob/main/doc/api.md#flakenix)
+of the IOGX manual.
```
Comment (koslambrou) [inline CONTRIBUTING.adoc:24]: I think you removed too much. I would keep basic explaination of how to enter a Nix shell using `nix develop` and how to setup the cache. I feel like the link you provided does not explain it adequatly enough (too general explanation). Also, there's no explaination on how to setup the cache which should really be upfront in the CONTRIBUTING.adoc.

Additionnaly, I like how the following paragraphs help the user get everything in order.

Comment (michaelpj) [reply CONTRIBUTING.adoc:24]: I admit, I didn't check the doc in iogx but I assumed it was at least as comprehensive. That said, I do agree that it's better to duplicate a bit here. It's hard enough to get people to click through to CONTIBUTING, if they search for "shell" and don't find anything they may just give up.

Comment (zeme-wana) [reply CONTRIBUTING.adoc:20]: I have created a separate document with comprehensive instructions on installing and setting up nix (including how to setup the cache) [here](https://github.com/input-output-hk/iogx/blob/main/doc/nix-setup-guide.md), and I have updated this CONTRIBUTING to point to that document and mention how to enter the shell. I think that document can now act as the SSOT, or do you still think I should duplicate some parts of it here?

Comment (koslambrou) [reply CONTRIBUTING.adoc:20]: Yes, some duplication will be necessary. At a minimum, we should keep the warning about the binary cache. So I would keep the following paragraph:

```
However you should pay attention to the output of `nix develop`: if you notice that you are _building_ GHC, it is likely that your caches have not been configured correctly.
```

Additionally, maybe the duplication part should be handled by the people responsible for repo. So I would say I'm fine with the current changes, but we will restructure this document to give more context to users.

**Pull request merged**